File: seed/csharp-model/trace/src/SeedTrace/V2/Problem/CreateProblemRequestV2.cs
using System.Text.Json.Serialization;
using SeedTrace;
using SeedTrace.V2;

#nullable enable

namespace SeedTrace.V2;

public class CreateProblemRequestV2
{
    [JsonPropertyName("problemName")]
    public string ProblemName { get; init; }

    [JsonPropertyName("problemDescription")]
    public ProblemDescription ProblemDescription { get; init; }

    [JsonPropertyName("customFiles")]
    public CustomFiles CustomFiles { get; init; }

    [JsonPropertyName("customTestCaseTemplates")]
    public List<TestCaseTemplate> CustomTestCaseTemplates { get; init; }

    [JsonPropertyName("testcases")]
    public List<TestCaseV2> Testcases { get; init; }

    [JsonPropertyName("supportedLanguages")]
    public HashSet<Language> SupportedLanguages { get; init; }

    [JsonPropertyName("isPublic")]
    public bool IsPublic { get; init; }
}

File: seed/php-sdk/trace/src/V2/Problem/Types/CreateProblemRequestV2.php
<?php

namespace Seed\V2\Problem\Types;

use Seed\Core\SerializableType;
use Seed\Core\JsonProperty;
use Seed\Problem\Types\ProblemDescription;
use Seed\Core\ArrayType;
use Seed\V2\Problem\Types\TestCaseTemplate;
use Seed\V2\Problem\Types\TestCaseV2;
use Seed\Commons\Types\Language;

class CreateProblemRequestV2 extends SerializableType
{
    #[JsonProperty("problemName")]
    /**
     * @var string $problemName
     */
    public string $problemName;

    #[JsonProperty("problemDescription")]
    /**
     * @var ProblemDescription $problemDescription
     */
    public ProblemDescription $problemDescription;

    #[JsonProperty("customFiles")]
    /**
     * @var mixed $customFiles
     */
    public mixed $customFiles;

    #[JsonProperty("customTestCaseTemplates"), ArrayType([TestCaseTemplate])]
    /**
     * @var array<TestCaseTemplate> $customTestCaseTemplates
     */
    public array $customTestCaseTemplates;

    #[JsonProperty("testcases"), ArrayType([TestCaseV2])]
    /**
     * @var array<TestCaseV2> $testcases
     */
    public array $testcases;

    #[JsonProperty("supportedLanguages"), ArrayType([Language])]
    /**
     * @var array<Language> $supportedLanguages
     */
    public array $supportedLanguages;

    #[JsonProperty("isPublic")]
    /**
     * @var bool $isPublic
     */
    public bool $isPublic;

    /**
     * @param string $problemName
     * @param ProblemDescription $problemDescription
     * @param mixed $customFiles
     * @param array<TestCaseTemplate> $customTestCaseTemplates
     * @param array<TestCaseV2> $testcases
     * @param array<Language> $supportedLanguages
     * @param bool $isPublic
     */
    public function __construct(
        string $problemName,
        ProblemDescription $problemDescription,
        mixed $customFiles,
        array $customTestCaseTemplates,
        array $testcases,
        array $supportedLanguages,
        bool $isPublic,
    ) {
        $this->problemName = $problemName;
        $this->problemDescription = $problemDescription;
        $this->customFiles = $customFiles;
        $this->customTestCaseTemplates = $customTestCaseTemplates;
        $this->testcases = $testcases;
        $this->supportedLanguages = $supportedLanguages;
        $this->isPublic = $isPublic;
    }
}
